rustls/src/client/hs.rs

```diff
@@ -600,7 +600,6 @@ impl State<ClientConnectionData> for ExpectServerHello {
                 randoms,
                 using_ems: self.using_ems,
                 transcript,
-                session_id: server_hello.session_id,
```

Comment:
@djc for historical reference, please write a more complete commit message. I suggest something like:

> it isn't obvious whether `tls12::CompleteServerHelloHandling::session_id` is the ID from the ClientHello or the ServerHello message (it is the latter). Eliminate the possibility for confusion by eliminating the field, since it is redundant with the `handle_server_hello`'s `server_hello` parameter.

Reply:
Added that. Sorry -- as mentioned in the other PR, I ran out of time yesterday. Also I didn't quite get to the point where I could articulate the benefit you're describing even though it was clearly what motivated me to make this change.